src/codeu/chat/client/core/View.java

```diff
@@ -92,6 +89,26 @@ public View(ConnectionSource source) {
   }
 
   @Override
+  public ServerInfo getServerInfo() {
+    try {
+      final Connection connection = this.source.connect();
+      Serializers.INTEGER.write(connection.out(), NetworkCode.SERVER_INFO_REQUEST);
+      if(Serializers.INTEGER.read(connection.in()) == NetworkCode.SERVER_INFO_RESPONSE) {
+        final Uuid version = Uuid.SERIALIZER.read(connection.in());
+        return new ServerInfo(version);
+      } else {
+        System.out.println("Oops, something wen't wrong. Please try again");
```

Comment:
A more descriptive message would be helpful. E.g., "Expected SERVER_INFO_RESPONSE but didn't receive it."